### ISSUE TITLE
Clearing queue doesn't clear song progress bar

### ISSUE THREAD

**Issue by badsketch** [nukeop/nuclear#536]
If a song is playing and the queue is cleared, the pink progress bar is still left over where instead it should disappear.

Gonna go ahead and fix it. I wonder if this is a regression? How come I feel like this wasn't happening before. Tried looking at commit history and don't see anything that might have affected it, but I didn't dig too deeply. Feel like I'm going crazy haha. I'll try to add some tests for it.

**Comment by nukeop** [comment]
I don't remember if this is a regression, but it's definitely a nice idea to fix it and ensure it stays that way with tests.

**Comment by badsketch** [comment]
Sorry this is taking so long.  😕 I'm gonna have a good chance to work on this during the weekend.

**Comment by nukeop** [comment]
No problem, take as much time as you need.

**Comment by badsketch** [comment]
Man, I have been struggling with writing appropriate tests for this for ages now. Hoping you can offer some pointers? I want to write an integration test that avoids testing implementation details so I've distilled the components to the basic ones.

This is to test that clearing the queue will reset the player to fill 0.
```
// create fake store with 50% playbackprogress
// mount queue menu more
// mount seekbar connected to playbackprogress
// simulate click clear queue
// expect seeekbar with 0% fill
```
What I have so far sort of looks like this

```javascript
const wrapper = mount(
    <Provider store={createFakeStore()}>
        <QueueMenuMore />
        {connect(state => ({ fill: state.playbackProgress }))(Seekbar)}
    </Provider>
)
```
But it's just not mounting correctly or mounting with errors, or maybe this simply isn't the correct approach to test a redux application? Wondering if there's any link you can provide to point me in the right direction. Hope that makes sense. Thanks for all the help. 🙂

**Comment by nukeop** [comment]
You can try using `shallow` instead of `mount`. I think a shallow wrapper will be fine for this test. Also note that for the current implementation of the seekbar, you need to also provide a queue prop.

You can also test the action, the reducer, and the component in separation, although you lose the integration part of the test in that case. I don't think there's much room for error if all three work as expected though. If you need a more complete mock store you can use a package called `redux-mock-store`.

**Comment by badsketch** [comment]
Hey, ended up getting it to work by extracting the connect to a separate line and using enzyme.update().

The issue I have now is I've created a thunk that calls the `sendPaused()` player action, which happens to use electrons ipcrenderer and is errors out as undefined in the tests. By chance, have you ever created a quick and dirty solution to mock it? I've taken a look at `mocha-electron` (getting weird errors right off the bat, might go back to it), and chai spies (can't spy on exported functions), but they're not working. My last hope I think is trying [Jest's moduleNameMapper](https://stackoverflow.com/questions/46898185/electron-jest-ipcrenderer-is-undefined-in-unit-tests) config option. I want to know if I go down that route, would you entertain the idea of letting me transition the app to Jest as the primary test runner? 😬

**Comment by nukeop** [comment]
Could you open a pull request draft with your current code so that I can better understand your problem?

It sounds to me like you could inject the ipc renderer either this way:
https://github.com/nukeop/nuclear/blob/master/packages/app/app/actions/playlists.injectable.js

Or using recompose's `withProps` HOC, and then test the base component without the HOC and supply your own mock ipc module. Anyway if you could

**Comment by badsketch** [comment]
Thanks for the suggestions! I'll take a look at those, and I also submitted a draft PR. 

Also, the reason there are additional commits other than my own in the PR is because the `feat/album-grid` branch is behind master and my branch is slightly ahead, correct?

**Comment by nukeop** [comment]
Btw, instead of grepping, you can use `.only()`: https://mochajs.org/#exclusive-tests

**Comment by badsketch** [comment]
Of course `.only()` works now after you suggest it 🤦‍♂️. I don't know what i was doing wrong earlier. Thanks for the suggestion! 🚀

**Comment by badsketch** [comment]
closed via #564
